docs/customresourcestate-metrics.md

```diff
@@ -272,19 +267,51 @@ spec:
         target_kind: [spec, targetRef, kind]
         target_name: [spec, targetRef, name]
       metrics:
-        - name: "annotations"
-          help: "Kubernetes annotations converted to Prometheus labels."
+        - name: "vpa_containerrecommendations_target"
```

Comment:
Suggest to follow the previous VPA metrics naming format "verticalpodautoscaler_status_recommendation_containerrecommendations_target_cpu". Append resource (cpu/memory) at the end of metrics name due to this is changed to be customized metrics name, we can't have duplicate metrics name at line 270 and line 282 which both are "vpa_containerrecommendations_target".

Reply:
@chihshenghuang I slightly disagree due to two reasons:
- The name of the metric will be super long as there is already a prefix appended automatically by the custom resource (check the doc further for the full name)
- The previous implemention had already the same metric with different labels / selectors for CPU / Memory and I think that to make it close to the original implementation (prior to v2.9.0), it should be like that.

Remember this is just an example and people can freely choose what to do / how to name the metric. I can suggest adding a line or two in the docs that this is also possible of course.

Reply:
using the same metric name both for cpu and memory will result in two HELP messages for the same type (ie `# HELP kube_customresource_vpa_containerrecommendations_target VPA container recommendations for memory` and `# HELP kube_customresource_vpa_containerrecommendations_target VPA container recommendations for cpu`). As a result, client (for example telegraf) fails to fetch metrics with `Error in plugin: error reading metrics for http://kube-state-metrics.kube-system.svc.cluster.local:8080/metrics: reading text format failed: text format parsing error in line 11485: second HELP line for metric name "kube_customresource_vpa_containerrecommendations_target"`.

So +1 for appending resource at the end (ie `- name: "vpa_containerrecommendations_target_cpu"`).
And if we do this, probably `commonLabels` can be removed